src/System.Runtime/tests/System/String.cs

```diff
@@ -1544,6 +1544,48 @@ public static void TestTrim()
 
         s = ". Foo .".TrimEnd('.');
         Assert.Equal(". Foo ", s);
+
+        s = "".Trim('c');
```

Comment:
I'll try to break this up and make it more readable. Yes, I'll add the test for array with single element.Thanks